[Yeo Yuan Xin] Duke Increments

yyuanxin opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

yyuanxin committed:
greet, echo, exit

yyuanxin committed:
add, lists

yyuanxin committed:
mark as done

yyuanxin committed:
todos, events, deadlines

yyuanxin committed:
handle errors

yyuanxin committed:
delete

yyuanxin committed:
save

yyuanxin committed:
dates and times

yyuanxin committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  dates and times

# Conflicts:
#	src/main/java/Duke.class
#	src/main/java/Duke.java

yyuanxin committed:
More OOP

yyuanxin committed:
Merge branch 'gradle' of https://github.com/yyuanxin/duke

* 'gradle' of https://github.com/yyuanxin/duke:
  Add sample checkstyle configuration
  Add support for Gradle workflow

yyuanxin committed:
JUnit testing

yyuanxin committed:
JAR file

yyuanxin committed:
javadoc

yyuanxin committed:
find tasks

yyuanxin committed:
coding standards

yyuanxin committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  find tasks

yyuanxin committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  coding standards

yyuanxin committed:
organize classes into java packages

yyuanxin committed:
update gradle - more test cases added

yyuanxin committed:
added checkstyle

yyuanxin committed:
GUI

yaaanch commented:
Your program is really well-structured. I think the OOP breakdown is extremely well done.

I think to further the OOP, one possibility would be to have more types of exceptions of different classes that extend DukeException. Then instead of changing the message used to construct the DukeException, objects from the classes with default messages could just be thrown.

It's great that you've written most of the Javadocs comments. For Javadocs comments, you may want to look at the style guide as well. So for example, capitalisation and starting functions with singular verbs. Another addition would be using the @ keywords such as @param and @return.

Overall, it's really well done!

yyuanxin committed:
Add assertions in methods

yyuanxin committed:
Improve Code Quality with JavaDocs and refactor codes

Add and Modify JavaDocs to align to coding standards
Replace some "if" statements to "switch" block
Make Parser Methods to static

yyuanxin committed:
Merge pull request #3 from yyuanxin/branch-A-Assertions

Add assertions in methods

yyuanxin committed:
Merge pull request #4 from yyuanxin/branch-A-CodeQuality

Improve Code Quality with JavaDocs and refactor codes

yyuanxin committed:
Implement friendlier syntax extension

Allow users to define their own aliases

Yoshi275 commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+            } else {
+                throw new AssertionError("ERROR: Invalid Task Type!");
+            }
+        } catch (DukeException e) {
```
Perhaps you don't need to `catch` the `DukeException` to throw it again? You can just `throw` the exception directly.

atharvjoshi commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,169 @@
+                type = Alias.aliases.get(subcommand);
+            } else {
+                throw new DukeException("       OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
```
Hello! Our group reviewed your code during our tutorial. Your parse method seems too long - perhaps you could abstract the splitting of the command and the matching of the aliases to another method called by parse()? Cheers!

meiannn commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,168 @@
+
+                switch (arr[0]) {
+                case "T": //todo
+                    task = new Todo(arr[2]);
```
Hi, you may want to consider extracting the checking of task type out, so that the switch block doesn't have to be repeated in many different classes

madanalogy commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,168 @@
+     * @param command String command from user input
+     * @param position Index of task in TaskList
+     * */
+    public void updateTasks(Task task, String command, int position) {
```
Hi, so we were reviewing your code during our tutorial, and do you think it might be better if you abstract out certain portions of this method into separate methods? Otherwise it's really good stuff great job!

yyuanxin committed:
Add Travis support

yyuanxin committed:
Improve GUI appearance
Add JavaFX CSS StyleSheet

yyuanxin committed:
Add product screenshot

yyuanxin committed:
Add User Guide

Update README.md

yyuanxin committed:
Set theme jekyll-theme-slate

yyuanxin committed:
Set theme jekyll-theme-slate

yyuanxin committed:
Add jar file for product release
Add shadowJar plugin to build.gradle
Add check for file does not exist in Alias class
Generalize text files directory for .jar release to work

yyuanxin committed:
Add cross-platform support

yyuanxin committed:
Fix Checkstyle error
Separate test cases into specific test classes

yyuanxin committed:
add full window resizing
package files to organize

yyuanxin committed:
Fix missing minutes of date time
Add more helpful error messages

yyuanxin committed:
Update User Guide

yyuanxin committed:
Improve code to update text file
Fix unable to overwrite bye command alias issue.

yyuanxin committed:
Add introduction and get started in user guide

yyuanxin committed:
Set theme jekyll-theme-cayman

yyuanxin committed:
Fix typo in exception message

yyuanxin committed:
Update User Guide

yyuanxin committed:
Fix overwrite alias in wrong order issue
